examples/asr/asr_chunked_inference/rnnt/speech_to_text_buffered_infer_rnnt.py

```diff
@@ -0,0 +1,282 @@
+
+# Middle Token merge algorithm
+
+python speech_to_text_buffered_infer_rnnt.py \
+    --asr_model="<Path to a nemo model>" \
+    --test_manifest="<Path to a JSON manifest>" \
+    --model_stride=4 \
+    --output_path="." \
+    --total_buffer_in_secs=10.0 \
+    --chunk_len_in_ms=8000 \
+    --device="cuda:0" \
+    --batch_size=128
+
+
+# Longer Common Subsequence (LCS) Merge algorithm
+
+python speech_to_text_buffered_infer_rnnt.py \
+    --asr_model="<Path to a nemo model>" \
+    --test_manifest="<Path to a JSON manifest>" \
+    --model_stride=4 \
+    --output_path="." \
+    --use_lcs_merge \
+    --lcs_alignment_dir=<OPTIONAL: Some path to store the LCS alignments> \
+    --total_buffer_in_secs=10.0 \
+    --chunk_len_in_ms=8000 \
+    --device="cuda:0" \
+    --batch_size=128
+
+# NOTE:
+
+    You can use `DEBUG=1 python speech_to_text_buffered_infer_rnnt.py ...` to print out the
+    ground truth text and predictions of the model.
+
+"""
+
+import copy
+import json
+import math
+import os
+from argparse import ArgumentParser
+
+import torch
+import tqdm
+from omegaconf import OmegaConf, open_dict
+
+import nemo.collections.asr as nemo_asr
+from nemo.collections.asr.metrics.wer import word_error_rate
+from nemo.collections.asr.parts.utils.streaming_utils import (
+    BatchedFrameASRRNNT,
+    LongestCommonSubsequenceBatchedFrameASRRNNT,
+)
+from nemo.utils import logging
+
+can_gpu = torch.cuda.is_available()
+
+
+def get_wer_feat(
+    mfst, asr, frame_len, tokens_per_chunk, delay, preprocessor_cfg, model_stride_in_secs, device, batch_size
+):
+    hyps = []
+    refs = []
+    audio_filepaths = []
+
+    with open(mfst, "r") as mfst_f:
+        print("Parsing manifest files...")
+        for l in mfst_f:
+            row = json.loads(l.strip())
+            audio_filepaths.append(row['audio_filepath'])
+            refs.append(row['text'])
+
+    with torch.inference_mode():
+        with torch.cuda.amp.autocast():
+            batch = []
+            asr.sample_offset = 0
+            for idx in tqdm.tqdm(range(len(audio_filepaths)), desc='Sample:', total=len(audio_filepaths)):
+                batch.append((audio_filepaths[idx], refs[idx]))
+
+                if len(batch) == batch_size:
+                    audio_files = [sample[0] for sample in batch]
+
+                    asr.reset()
+                    asr.read_audio_file(audio_files, delay, model_stride_in_secs)
+                    hyp_list = asr.transcribe(tokens_per_chunk, delay)
+                    hyps.extend(hyp_list)
+
+                    batch.clear()
+                    asr.sample_offset += batch_size
+
+            if len(batch) > 0:
+                asr.batch_size = len(batch)
+                asr.frame_bufferer.batch_size = len(batch)
+                asr.reset()
+
+                audio_files = [sample[0] for sample in batch]
+                asr.read_audio_file(audio_files, delay, model_stride_in_secs)
```

Comment:
Hmm it can be either way. But this way we can try out different values of stride without rebuilding the buffer processor. Lets keep it as is to remain close to CTC implementation.